pkg/kubelet/eviction/helpers.go

```diff
@@ -1234,14 +1234,18 @@ func evictionMessage(resourceToReclaim v1.ResourceName, pod *v1.Pod, stats stats
 	if quantity != nil && available != nil {
 		message += fmt.Sprintf(thresholdMetMessageFmt, quantity, available)
 	}
-	containers := []string{}
+	exceededContainers := []string{}
 	containerUsage := []string{}
 	podStats, ok := stats(pod)
 	if !ok {
 		return
 	}
+	containers := pod.Spec.Containers
+	if len(pod.Spec.InitContainers) != 0 {
+		containers = append(containers, pod.Spec.InitContainers...)
```

Comment:
oh, I see. there is a comparison with requests and there is no requests for ephemeral contianers. Can you please add this comment here explicitly so next person looking at this code will know why we are not accounting for ephemeral container here.

Reply:
Thanks! As for the ephemeral containers, it was simply my oversight. A you mentioned, specifying the `resources` field results in an error during API server validation, so resource requests can't be set for ephemeral containers. 
Added a comment explaining why ephemeral containers aren’t included here.
https://github.com/kubernetes/kubernetes/blob/f0f7ff989a948389247e628c4c5a43e915f51daa/pkg/apis/core/validation/validation.go#L88

```console
❯ cat <<EOF | kind create cluster --config -
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
  image: kindest/node:v1.31.0@sha256:53df588e04085fd41ae12de0c3fe4c72f7013bba32a20e7325357a1ac94ba865
- role: worker
  image: kindest/node:v1.31.0@sha256:53df588e04085fd41ae12de0c3fe4c72f7013bba32a20e7325357a1ac94ba865
EOF
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.31.0) 🖼
 ✓ Preparing nodes 📦 📦
 ✓ Writing configuration 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
 ✓ Joining worker nodes 🚜
Set kubectl context to "kind-kind"
You can now use your cluster with:

kubectl cluster-info --context kind-kind

Have a nice day! 👋

❯ kubectl version
Client Version: v1.31.1
Kustomize Version: v5.4.2
Server Version: v1.31.0

❯ cat <<'EOF' | kubectl create -f -
apiVersion: v1
kind: Pod
metadata:
  name: alloc
spec:
  terminationGracePeriodSeconds: 1
  restartPolicy: Never
  containers:
  - name: el1
    image: registry.k8s.io/e2e-test-images/agnhost:2.52
    command: ["sleep", "infinity"]
EOF
pod/alloc created

❯ cat <<EOF > custom.json
{
  "resources": {
    "requests": {
      "cpu": "0.2",
      "memory": "256Mi"
    }
  }
}
EOF

❯ kubectl debug alloc -it --custom custom.json --image=registry.k8s.io/e2e-test-images/agnhost:2.52 --target=el1 -- sh
Targeting container "el1". If you don't see processes from this container it may be because the container runtime doesn't support this feature.
Defaulting debug container name to debugger-ch2t6.
The Pod "alloc" is invalid: spec.ephemeralContainers[0].resources: Forbidden: cannot be set for an Ephemeral Container
```

Reply:
cool, thx